File: area.cpp
#include <iostream>
using namespace std;

int main() {

char ch;
char shape;
float result;
float a, b, r, s, bt, ht, w, lb, ub, htt, th, rs, ae, be, qb, rc, hc;

print:
    cout << "For 2D Shapes:\nUse 'r' for rectangle,\nUse 'c' for circle,\nUse 's' for square\nUse 't' for triangle,\nUse 'w' for trapezium,\nUse 'v' for sector,\nUse 'e' for ellipse";
    cout << "\n\nFor 3D Shapes:\nUse 'q' for cube,\nUse 'y' for TSA of cylinder\n";

    cout << "\nEnter shape to find the area: ";
    cin >> shape;

    switch(shape) {
        case 'r' :
        cout << "\nCalculating the area of a rectangle\n";
        cout << "\nEnter length of the rectangle: ";
        cin >> a;
        cout << "\nEnter width of the rectangle: ";
        cin >> b;
        result = a * b;
        cout << "\nArea of the rectangle is: " <<   result << " Unit^2";
        break;
    
        case 'c' :
        cout << "\nCalculating the area of a circle\n";
        cout << "\nEnter radius of the circle: ";
        cin >> r;
        result = 3.1415926535 * r * r;
        cout << "\nArea of the circle is: " <<  result << " Unit^2";
        break;
    
        case 's' :
        cout << "\nCalculating the area of a square\n";
        cout << "\nEnter side of the square: ";
        cin >> s;
        result = s * s;
        cout << "\nArea of the square is: " <<  result << " Unit^2";
        break;
    
        case 't' :
        cout << "\nCalculating the area of a triangle\n";
        cout << "\nEnter base of the triangle: ";
        cin >> bt;
        cout << "\nEnter height of the triangle: ";
        cin >> ht;
        result = (ht * bt) / 2;
        cout << "\nArea of the triangle is: " <<    result << " Unit^2";
        break;
    
        case 'w' :
        cout << "\nCalculating the area of a trapezium\n";
        cout << "\nEnter lower base of the trapezium: ";
        cin >> lb;
        cout << "\nEnter upper base of the trapezium: ";
        cin >> ub;
        cout << "\nEnter height of the trapezium: ";
        cin >> htt;
        result = ((lb + ub) / 2) * htt;
        cout << "\nArea of the trapezium is: " <<   result << " Unit^2";
        break;

        case 'v' :
        cout << "\nCalculating the area of a sector\n";
        cout << "\nEnter the angle subtended by the sector: ";
        cin >> th;
        cout << "\nEnter the radius of the sector: ";
        cin >> rs;
        result = (th / 360) * 3.1415926535 * rs * rs;
        cout << "\nArea of the sector is: " <<  result << " Unit^2";
        break;

        case 'e' :
        cout << "\nCalculating the area of an ellipse\n";
        cout << "\nEnter the radius at minor axis: ";
        cin >> ae;
        cout << "\nEnter the radius at major axis: ";
        cin >> be;
        result = 3.1415926535 * ae * be;
        cout << "\nArea of the ellipse is: " <<     result << " Unit^2";
        break;

        case 'q' :
        cout << "\nCalculating the surface area of a cube\n";
        cout << "\nEnter the side of the cube: ";
        cin >> qb;
        result = 6 * (qb * qb);
        cout << "\nArea of the cube is: " <<    result << " Unit^2";
        break;

        case 'y' :
        cout << "\nCalculating the total surface area of a cylinder\n";
        cout << "\nEnter the radius of the cylinder: ";
        cin >> rc;
        cout << "\nEnter the height of the cylinder: ";
        cin >> hc;
        result = 2 * 3.1415926535 * rc * (rc + hc);
        cout << "\nTotal surface area of the cylinder is: " <<  result << " Unit^2";
        break;

    default : cout << "\nPlease enter a valid character for the corresponding shape!";

}

again:
    cout << "\n\nDo you want to repeat the operation? (Y/N) ";
    cin >> ch;
    if(ch == 'y' || ch == 'Y') {
    goto print;
    }

    else if(ch == 'n' || ch == 'N') {
    cout << "\nExiting the program...\n";
    cout << "\nDone!\n\n";
    }

    else {
    cout << "\nPlease enter Yes or No!";
    goto again;
    }

    return 0;
    
}

File: area.py
print("For 2D Shapes:")
print("Use 'r' for rectangle,")
print("Use 'c' for circle,")
print("Use 's' for square")
print("Use 't' for triangle,")
print("Use 'w' for trapezium,")
print("Use 'v' for sector,")
print("Use 'e' for ellipse")

print("\nFor 3D Shapes:")
print("Use 'q' for cube,")
print("Use 'y' for TSA of cylinder")

shape = input("\nEnter shape to find the area: ")

if shape == 'r':
    print("\nCalculating the area of a rectangle\n")
    a = float(input("Enter length of the rectangle: "))
    b = float(input("Enter width of the rectangle: "))
    result = a * b
    print(f"\nArea of the rectangle is: {result} Unit^2")
    

elif shape == 'c':
    print("\nCalculating the area of a circle\n")
    r = float(input("Enter radius of the circle: "))
    result = 3.1415926535 * r * r
    print(f"\nArea of the circle is: {result} Unit^2")

elif shape == 's':
    print("\nCalculating the area of a square\n")
    s = float(input("Enter side of the square: "))
    result = s * s
    print(f"\nArea of the square is: {result} Unit^2")

elif shape == 't':
    print("\nCalculating the area of a triangle\n")
    bt = float(input("Enter base of the triangle: "))
    ht = float(input("Enter height of the triangle: "))
    result = (ht * bt) / 2
    print(f"\nArea of the triangle is: {result} Unit^2")

elif shape == 'w':
    print("\nCalculating the area of a trapezium\n")
    lb = float(input("Enter lower base of the trapezium: "))
    ub = float(input("Enter upper base of the trapezium: "))
    htt = float(input("Enter height of the trapezium: "))
    result = ((lb + ub) / 2) * htt
    print(f"\nArea of the trapezium is: {result} Unit^2")

elif shape == 'v':
    print("\nCalculating the area of a sector\n")
    th = float(input("Enter the angle subtended by the sector: "))
    rs = float(input("Enter the radius of the sector: "))
    result = (th / 360) * 3.1415926535 * rs * rs
    print(f"\nArea of the sector is: {result} Unit^2")

elif shape == 'e':
    print("\nCalculating the area of an ellipse\n")
    ae = float(input("Enter the radius at minor axis: "))
    be = float(input("Enter the radius at major axis: "))
    result = 3.1415926535 * ae * be
    print(f"\nArea of the ellipse is: {result} Unit^2")

elif shape == 'q':
    print("\nCalculating the surface area of a cube\n")
    qb = float(input("Enter the side of the cube: "))
    result = 6 * qb * qb
    print(f"\nSurface area of the cube is: {result} Unit^2")

elif shape == 'y':
    print("\nCalculating the TSA of a cylinder\n")
    r = float(input("Enter the radius of the cylinder: "))
    h = float(input("Enter the height of the cylinder: "))
    result = 2 * 3.1415926535 * r * h + 2 * 3.1415926535 * r * r
    print(f"\nTSA of the cylinder is: {result} Unit^2")

else:
    print("\nInvalid shape!")
